data-preprocessing/webtextbook_parsing_script/extract_textbook_from_url_notebook_.ipynb

```diff
@@ -0,0 +1,462 @@
+    "\n",
+    "def extract_url(link, pageList, maxNmber, char = []):\n",
+    "    # This function generates the dirst number or alphabet that\n",
+    "    # is part of the table of content requied\n",
+    "    #------------------------------------------------------------------------#\n",
+    "\n",
+    "    #-> maxNumber = maximum number of the index numbers of the table of content\n",
+    "    #--> char =  alphabets or string index in the table of content\n",
+    "    # --> pageList = a list containing landing pages of all needed charpter(eg: [\"1-introduction\", 'chapter-2' .. ])\n",
+    "    # ---> link = url of the site excluding landing page indexing (eg: \"https://openstax.org/books/university-physics-volume-3/pages/\")\n",
+    "\n",
+    "    #------------------------------------------------------------------------#\n",
+    "    url = []\n",
+    "    list_pages = list(range(1,maxNmber+1))\n",
+    "    for item in range(len(char)):\n",
+    "        list_pages.append(char[item])\n",
+    "\n",
+    "    for item in pageList:\n",
+    "        for value in list_pages:\n",
+    "        \n",
+    "            if item.startswith(str(value)):\n",
+    "                url.append(link+item)\n",
+    "\n",
+    "    return  unique(url)\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Extract contents from a URL\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "def extract_url_content(url, file_name=''):\n",
+    "    # Send a GET request to the URL\n",
+    "    response = requests.get(url)\n",
+    "\n",
+    "    # Parse the HTML content\n",
+    "    soup = BeautifulSoup(response.content, \"html.parser\")\n",
+    "\n",
+    "    # Find the main content section\n",
+    "    main_content = soup.find(\"div\")\n",
+    "\n",
+    "   \n",
+    "    if main_content is None:\n",
+    "        print(\"Unable to find the main content section\")\n",
+    "        return\n",
+    "\n",
+    "    # Find the first title on the website\n",
+    "    titles = main_content.find_all([\"h1\", \"h2\", \"h3\", \"h4\", \"h5\"])\n",
+    "    first_title = None\n",
+    "    for title in titles:\n",
+    "        if title.text.strip():\n",
+    "            first_title = title.text.strip()\n",
+    "            first_title = str(file_name)+'-'+ first_title\n",
+    "            break\n",
+    "    file_name = first_title if first_title else str(file_name)+'-'+\"content\"\n",
+    "\n",
+    "    # List to store the content\n",
+    "    content_list = []\n",
+    "\n",
+    "     #----------------------------\n",
+    "     # Title of the chapter\n",
+    "    Heading = {}\n",
+    "    head = soup.find('head')\n",
+    "    Title = head.find_all('title')\n",
+    "    \n",
+    "\n",
+    "    for head in Title:\n",
```

Comment:
isn't head already used as a variable name on line 183? https://github.com/nsmq-ai/nsmqai/pull/77/files#diff-fd416dfc4e72ec4d6c15a8b7b9552d197f50f6a43ea91ccfb96eef02141ef17eR183
not sure it should be reused here again unless there's a good reason why